docs/crystal_design.md

```diff
@@ -0,0 +1,187 @@
+# Metadata Management (Crystal)
+
+**Author(s)**: [Pravin Ranjan](https://github.com/PravinRanjan10)
+
+## Motivation and background
+
+Metadata gives basic information about data—including type, time of creation, size of object/buckets, storage class, and more for cloud object storages. Metadata management is the business discipline of managing the metadata about data. It can play key roles during decision making activity. Thus metadata management is required nowadays.
+
+A user may provision s3-cloud buckets using some solution like strato(multi-cloud project from SODAFOUNDATION) and/or manually. To get and manage metadata of all buckets/objects provisioned by manually and automatically, another solution required. Because, there would be no metadata available for manually provisioned buckets.
```

Comment:
Why only s3-cloud buckets? Say cloud buckets that support S3 interface. Otherwise, people may think that we are competing with Amazon's metadata store. The design should be general. In the implementation, we could say that currently, S3 buckets interface is supported, but also clarify that many cloud systems support this interface